plugin/trino-faker/src/main/java/io/trino/plugin/faker/FakerPageSource.java

```diff
@@ -257,8 +261,10 @@ private Generator constraintedValueGenerator(FakerColumnHandle handle, Domain do
         };
     }
 
-    private Generator randomValueGenerator(FakerColumnHandle handle, Range range)
+    private Generator randomValueGenerator(FakerColumnHandle handle, Range predicateRange)
     {
+        Range range = predicateRange.intersect(handle.range())
+                .orElseThrow(() -> new TrinoException(INVALID_ROW_FILTER, "Predicates do not overlap with column min and max properties"));
```

Comment:
Why is this an error ? Can't we generate empty page for this case ?

Reply:
This is removed in the following commit, so it doesn't matter how we handle it here.